src/components/Search/Filter.tsx

```diff
@@ -0,0 +1,298 @@
+import React, { ReactElement, useEffect } from 'react'
+import classNames from 'classnames/bind'
+import { addExistingParamsToUrl } from './utils'
+import Button from '@shared/atoms/Button'
+import {
+  FilterByAccessOptions,
+  FilterByTypeOptions
+} from '../../@types/aquarius/SearchQuery'
+import { useRouter } from 'next/router'
+import queryString from 'query-string'
+import styles from './Filter.module.css'
+import { useFilter, Filters, Sort } from '@context/Filter'
+import Input from '@components/@shared/FormInput'
+import Accordion from '@components/@shared/Accordion'
+
+const cx = classNames.bind(styles)
+
+interface FilterStructure {
+  id: keyof Filters
+  label: string
+  type: string
+  options: {
+    label: string
+    value: string
+  }[]
+}
+
+const filterList: FilterStructure[] = [
+  {
+    id: 'serviceType',
+    label: 'Service Type',
+    type: 'filterList',
+    options: [
+      { label: 'datasets', value: FilterByTypeOptions.Data },
+      { label: 'algorithms', value: FilterByTypeOptions.Algorithm }
+    ]
+  },
+  {
+    id: 'accessType',
+    label: 'Access Type',
+    type: 'filterList',
+    options: [
+      { label: 'download', value: FilterByAccessOptions.Download },
+      { label: 'compute', value: FilterByAccessOptions.Compute }
+    ]
+  },
+  {
+    id: 'filterSet',
+    label: 'Categories',
+    type: 'filterList',
+    options: [
+      {
+        label: 'automotive',
+        value: 'automotive'
+      },
+      {
+        label: 'manufacturing',
+        value: 'manufacturing'
+      },
+      {
+        label: 'text analysis',
+        value: 'textAnalysis'
+      },
+      {
+        label: 'finance',
+        value: 'finance'
+      }
+    ]
+  }
+]
+
+export const filterSets = {
+  automotive: [
+    'charging',
+    'ev',
+    'gx4m',
+    'mobility',
+    'moveid',
+    'parking',
+    'traffic'
+  ],
+  manufacturing: [
+    'euprogigant',
+    'industry40',
+    'manufacturing',
+    'predictive-maintenance'
+  ],
+  textAnalysis: ['library', 'ocr', 'text-analysis'],
+  finance: ['graphql']
+}
```

Comment:
suggestion: the same applies for these set definitions

maybe a config file just to specify filter-sets makes sense at that point (`const filterSets` and `id: 'filterSet'` could be moved there)